build-caching-maven-samples/enforcer-project/pom.xml

```diff
@@ -128,118 +137,31 @@
                           <ignore>mojoExecution</ignore>
                         </ignoredProperties>
                       </inputs>
-                      <iteratedProperties>
-                        <property>
-                          <name>rules</name>
-                          <inputs>
-                            <properties>
-                              <property>
-                                <name>includes</name>
-                              </property>
-                              <property>
-                                <name>excludes</name>
-                              </property>
-                              <property>
-                                <name>searchTransitive</name>
-                              </property>
-                              <property>
-                                <name>message</name>
-                              </property>
-                            </properties>
-                          </inputs>
-                        </property>
-                      </iteratedProperties>
-                      <nestedProperties>
-                        <property>
-                          <name>project</name>
-                          <iteratedProperties>
-                            <property>
-                              <name>dependencies</name>
-                              <inputs>
-                                <properties>
-                                  <property>
-                                    <name>groupId</name>
-                                  </property>
-                                  <property>
-                                    <name>artifactId</name>
-                                  </property>
-                                  <property>
-                                    <name>version</name>
-                                  </property>
-                                  <property>
-                                    <name>type</name>
-                                  </property>
-                                  <property>
-                                    <name>scope</name>
-                                  </property>
-                                </properties>
-                              </inputs>
-                            </property>
-                          </iteratedProperties>
-                        </property>
-                      </nestedProperties>
                       <outputs>
                         <cacheableBecause>banned dependency check should run when dependencies change</cacheableBecause>
                       </outputs>
                     </execution>
                     <execution>
                       <id>enforce-bytecode-version</id>
                       <inputs>
+                        <properties>
+                          <property>
+                            <name>rules</name>
```

Comment:
My understanding is that `rules` of type [PlexusConfiguration](https://github.com/apache/maven-enforcer/compare/enforcer-3.1.0...enforcer-3.2.1#diff-376d0eba74cd031bde69018690488d25360994a47cb7b5cf962388fc119a551eR137) can't be defined as a property. Quoting the [doc](https://docs.gradle.com/enterprise/maven-extension/#making_other_goals_cacheable):

> For input properties, the extension supports all primitives, Strings, Enums and Collections, Arrays and Maps of those. Any other types need to be broken down using the nestedProperties (for a single complex type) or iteratedProperties (for a Collection of complex types) configuration.

Reply:
👍